library/src/schemas/enum/enum.ts

```diff
@@ -94,7 +119,7 @@ export function enum_(
   message?: ErrorMessage<EnumIssue>
 ): EnumSchema<Enum, ErrorMessage<EnumIssue> | undefined> {
   const options = Object.entries(enum__)
-    .filter(([key]) => isNaN(+key))
+    .filter(([key]) => (+key).toString() !== key)
```

Comment:
In my tests, your change did not solve the `v.enum({ 1: "A" })` problem. My recommendation would be as follows:

```ts
return Object.entries(enum__)
    .filter(([key, value]) => isNaN(+key) || enum__[value] === undefined)
    .map(([, value]) => value);
```

The additional `enum__[value] === undefined` check ensures that only reverse mapping entries are removed.

Regarding the `+key` issue you mentioned. Can you provide an example that results in a bug with the current implementation?

Reply:
I further investigated this PR and the following code seems to fix all cases mentioned above. Should we add this code to this PR and revert the `FilterEnumKeys` changes?

```ts
Object.entries(enum__)
  .filter(([key, value]) => `${+key}` !== key || enum__[value] === undefined)
  .map(([, value]) => value);
```

Reply:
@fabian-hiller I refined the `enum__[value] === undefined` part. Is this better?

```
  const options = Object.entries(enum__)
    .filter(
      ([key, value]) =>
        !(
          `${+key}` === key &&
          typeof value === 'string' &&
          Object.is(enum__[value], +key)
        )
    )
    .map(([, value]) => value);
```

Reply:
Yes, this looks good to me. Should I take over and finalize the PR or do you want to add this change?